### ISSUE TITLE
🐛 disabled linter doesn't show errors but autofixes them anyways

### ISSUE THREAD

**Issue by Akuukis** [biomejs/biome-vscode#491]
### VS Code version

1.96.2

### Extension version

2.3.2

### Biome version

1.9.4

### Operating system

- [ ] Windows
- [ ] macOS
- [x] Linux

### Description

In VSCode I don't see linter errors when linter is disabled but biome fixes them anyways, e.g. this rule https://biomejs.dev/linter/rules/use-arrow-function/

With the following biome config
```jsonc
{
  ...
  "linter": {
    "enabled": false // TODO: for later.
  }
}
```
and the following vscode workspace config
```jsonc
{
  ...
  "settings": {
    ...
    "[javascript]": {
      "editor.defaultFormatter": "biomejs.biome",
      "editor.codeActionsOnSave": {
        "source.fixAll.biome": "explicit"
      }
    },
  }
}
```

### Steps to reproduce

1. set settings as above
2. restart VSCode just in case
3. open any `.js` file with a function that should be arrow function instead
4. notice that biome extension doesn't highlight it
5. save the file
6. notice that biome extension did auto-fix it

### Expected behavior

Don't autofix linter rules when linter is disabled.
Also in other words, fix only what's displayed as fixable.

### Does this issue occur when using the CLI directly?

No

### Logs

```plain text
# Biome
[Info  - 1:46:09 PM] Server initialized with PID: 72361

# Biome LSP (/)
2025-02-07 13:45:05.952 [info] [Info  - 1:45:05 PM] Server initialized with PID: 72361
```

**Comment by nhedger** [comment]
Please provide a **minimal** reproduction repository

**Comment by Akuukis** [comment]
@nhedger @shulandmimi please see the minimal reproduction repository here https://github.com/Akuukis/biome-repro

**Comment by Akuukis** [comment]
@ematipico thanks, I changed it (see [commit](https://github.com/Akuukis/biome-repro/commit/0ccdb647fffaca61702cc47ceacd148fcac6992c)) but the bug still stands as described.

**Comment by nhedger** [comment]
This has been fixed for some time inside of Biome. I cannot pinpoint a specific version, but i cannot reproduce in the latest stable version, while I was able to reproduce using the version of Biome specific in the repro's package.json.
